airflow/www/views.py

```diff
@@ -2288,6 +2291,38 @@ class UserModelView(wwwutils.SuperUserMixin, AirflowModelView):
     column_default_sort = 'username'
 
 
+class VersionView(wwwutils.SuperUserMixin, BaseView):
+    @expose('/')
+    def version(self):
+        # Look at the version from setup.py
+        airflow_version = None
+        try:
+            airflow_version = pkg_resources.require("airflow")[0].version
+        except :
+            pass
+
+        # Get the Git repo and git hash : first try pip freeze, then look at git
+        git_proj_uri = git_repo = git_hash = None
+        # Try looking at Git
+        try:
+            repo = Repo(os.path.expanduser(settings.AIRFLOW_HOME))
+            head_commit = repo.head.commit
+            git_hash = head_commit.hexsha
+            git_repo=repo.remotes.origin.url
+        except:
+            pass
+
+        # Render information
+        title = "Version Info"
+        if git_repo:
+            git_proj_uri = git_repo.replace('git@','').replace('.git','').replace(":","/")
```

Comment:
This seems a bit brittle as there are a lot of different ways a remote could be displayed:
http://stackoverflow.com/questions/2514859/regular-expression-for-git-repository

Also as Max mentioned, origin might not exist or it might not be the remote that the commit lives on even if it does exist, which could be confusing since the URL generated will 404. I think just listing the SHA without a URL is perfectly fine, LMK what you think.